src/LibraryManager.Vsix/Json/Completion/VersionCompletionEntry.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.Text;
+using Microsoft.Web.Editor.Completion;
+using Microsoft.Web.LibraryManager.Providers.Unpkg;
+
+namespace Microsoft.Web.LibraryManager.Vsix
+{
+    internal class VersionCompletionEntry : SimpleCompletionEntry
+    {
+        internal SemanticVersion SemVersion { get; private set; }
+
+        public VersionCompletionEntry(string displayText, string insertionText, string description, ImageMoniker moniker, ITrackingSpan span, IIntellisenseSession session, int specificVersion)
+            : base(displayText, insertionText, description, moniker, span, session, specificVersion)
+        {
+            if (!string.IsNullOrEmpty(displayText))
+            {
+                SemVersion = SemanticVersion.Parse(displayText);
+            }
+        }
+
+        protected override int InternalCompareTo(CompletionEntry other)
+        {
+            VersionCompletionEntry otherEntry = other as VersionCompletionEntry;
+
+            if (otherEntry == null)
+            {
+                return 1;
+            }
+
+            int result = CompareSemanticVersion(SemVersion, otherEntry.SemVersion);
```

Comment:
If OtherEntry is null, the self completion entry would win this comparison. So it can return 1. Remain this check here make sense?

---
In reply to: [205558481](https://github.com/aspnet/LibraryManager/pull/312#discussion_r205558481) [](ancestors = 205558481)

Reply:
Sorry, not clear here. It's not a huge deal, I just don't think the if condition before is necessary if you use ?. here.

---
In reply to: [205566579](https://github.com/aspnet/LibraryManager/pull/312#discussion_r205566579) [](ancestors = 205566579,205558481)

Reply:
I see. I will make a change.

---
In reply to: [205584519](https://github.com/aspnet/LibraryManager/pull/312#discussion_r205584519) [](ancestors = 205584519,205566579,205558481)